refactor(kosync): introduce differentiated sync strategies

zeedif opened the pull request:
This PR refactors the KOReader sync strategy to provide more granular control over conflict resolution.

Previously, a single strategy setting was applied universally. Now, users can define separate strategies for two distinct scenarios:
1.  **Forward Sync:** When the remote progress is newer than the local progress.
2.  **Backward Sync:** When the remote progress is older than the local progress.

This is achieved by deprecating the old `koreaderSyncStrategy` and introducing two new settings: `koreaderSyncStrategyForward` and `koreaderSyncStrategyBackward`.

The `SILENT` option has been removed from the strategy enum because its behavior is now implicitly defined by the user's choices. For example:
- Setting `Forward` to `KEEP_REMOTE` and `Backward` to `KEEP_LOCAL` perfectly replicates the old `SILENT` behavior (always sync to the most recent timestamp).

All previous functionalities are retained and new, more complex combinations are now possible, giving users greater customization. Existing user settings are automatically migrated to the new system for a smooth transition.

zeedif committed:
refactor(kosync): introduce differentiated sync strategies

Replaces the single `koreaderSyncStrategy` setting with `koreaderSyncStrategyForward` and `koreaderSyncStrategyBackward`. This allows users to define distinct conflict resolution behaviors based on whether the remote progress is newer or older than the local progress.

The `KoreaderSyncStrategy` enum has been simplified to `KoreaderSyncConflictStrategy` with four clear options: `PROMPT`, `KEEP_LOCAL`, `KEEP_REMOTE`, and `DISABLED`. The ambiguous `SILENT` option is removed, as its behavior is now implicitly covered by selecting `KEEP_REMOTE` for forward syncs and `KEEP_LOCAL` for backward syncs.

The legacy `koreaderSyncStrategy` setting is now deprecated and is seamlessly migrated to the new dual-strategy system using `MigratedConfigValue`, ensuring backward compatibility for existing user configurations.

Syer10 commented on server/server-config/src/main/kotlin/suwayomi/tachidesk/server/ServerConfig.kt:
New settings need to be added to the bottom of the serverconfig and given the next proto number slot, they can't be added in the middle.

zeedif committed:
fix(kosync): correct proto numbers and setting order for sync strategies

zeedif committed:
fix(kosync): proto number 78 to 68

schroda commented on server/server-config/src/main/kotlin/suwayomi/tachidesk/server/ServerConfig.kt:
```diff
@@ -599,12 +600,58 @@ class ServerConfig(
 
-    val koreaderSyncStrategy: MutableStateFlow<KoreaderSyncStrategy> by EnumSetting(
+    @Deprecated("Use koreaderSyncStrategyForward and koreaderSyncStrategyBackward instead")
+    val koreaderSyncStrategy: MutableStateFlow<KoreaderSyncLegacyStrategy> by MigratedConfigValue(
```
I just noticed that I've missed https://github.com/Suwayomi/Suwayomi-Server/blob/master/server/src/main/kotlin/suwayomi/tachidesk/server/ServerSetup.kt#L310 in the settings cleanup, so you'd have to add the migration logic here as well for now.
Otherwise, the current setting gets lost on a server start.

zeedif committed:
fix(server): migrate KOReader sync strategy during settings cleanup

Add migration logic to convert the old `server.koreaderSyncStrategy` key
into the new `server.koreaderSyncStrategyForward` and
`server.koreaderSyncStrategyBackward` keys during server setup.
